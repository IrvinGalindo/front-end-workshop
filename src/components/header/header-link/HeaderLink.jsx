import React from 'react';
import PropTypes from 'prop-types';

const HeaderLink = ({ onClick, label }) => (
  <li className="yta_Header-menu-list-item">
    <button type="button" onClick={onClick} className="yta_Link-underline--primary">
      {label}
    </button>
  </li>
);

HeaderLink.propTypes = {
  onClick: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
};

export default HeaderLink;
