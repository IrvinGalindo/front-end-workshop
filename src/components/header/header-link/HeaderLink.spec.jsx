import React from 'react';
import { shallow } from 'enzyme';

import HeaderLink from './HeaderLink';

describe('HeaderLink', () => {
  const fakeEvent = jest.fn();
  const headerLink = shallow(<HeaderLink onClick={fakeEvent} label="fake link" />);

  it('should render', () => {
    expect(headerLink).toMatchSnapshot();
  });

  it('should have been fakeEvent called', () => {
    headerLink.find('button').simulate('click');
    expect(fakeEvent).toHaveBeenCalled();
  });
});
