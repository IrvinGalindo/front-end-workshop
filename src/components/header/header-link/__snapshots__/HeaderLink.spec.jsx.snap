// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLink should render 1`] = `
<li
  className="yta_Header-menu-list-item"
>
  <button
    className="yta_Link-underline--primary"
    onClick={[MockFunction]}
    type="button"
  >
    fake link
  </button>
</li>
`;
